great_expectations/rule_based_profiler/config/base.py

```diff
@@ -408,29 +406,23 @@ def resolve_config_using_acceptable_arguments(
     ) -> "RuleBasedProfilerConfig":  # noqa: F821
         runtime_config: RuleBasedProfilerConfig = profiler.config
 
-        runtime_variables: Optional[
-            ParameterContainer
-        ] = profiler.reconcile_profiler_variables(variables)
-        runtime_variables_configs: dict = (
-            runtime_variables.to_dict()["parameter_nodes"]["variables"]["variables"]
-            or {}
-        )
-
-        effective_rules: List["Rule"] = profiler.reconcile_profiler_rules(  # noqa: F821
-            rules=rules
+        runtime_variables: dict = profiler.reconcile_profiler_variables_as_dict(
+            variables=variables
         )
 
-        rule: "Rule"  # noqa: F821
+        effective_rules: Dict[
+            str, "Rule"  # noqa: F821
+        ] = profiler.reconcile_profiler_rules_as_dict(rules=rules)
         runtime_rules: Dict[str, dict] = {
-            rule.name: rule.to_json_dict() for rule in effective_rules
+            name: rule.to_dict() for name, rule in effective_rules.items()
```

Comment:
This was all changed in a prior PR but we should be leveraging the `as_dict` method as opposed to the full reconciliation.

Reply:
@cdkini My preference is to always use the highest possible level method (within reason).  Here, by using the higher level method, we can -- in the future -- pass the non-default reconciliation directives.  There is no need to leverage "as_dict" -- it is really a "protected" method (that does not exist in Python, and we did not want to make it private, because it is used outside of the module, where it is defined).  Happy to discuss.  Thank you.

Reply:
to consider implications of both approach - both have value but we need to see what makes the most sense here.